outlookmail/tests/test.bal

```diff
@@ -0,0 +1,414 @@
+import ballerina/log;
+import ballerina/os;
+import ballerina/test;
+import ballerina/io;
+
+configurable string & readonly refreshUrl = os:getEnv("REFRESH_URL");
+configurable string & readonly refreshToken = os:getEnv("REFRESH_TOKEN");
+configurable string & readonly clientId = os:getEnv("CLIENT_ID");
+configurable string & readonly clientSecret = os:getEnv("CLIENT_SECRET");
+
+Configuration configuration = {
+    clientConfig: {
+        refreshUrl: refreshUrl,
+        refreshToken: refreshToken,
+        clientId: clientId,
+        clientSecret: clientSecret
+    }
+};
+
+Client oneDriveClient = check new (configuration);
+string messageId = "";
+string createdDraftId = "";
+string sentMessageId = "";
+string mailFolderId = "";
+string searchMailFolderId = "";
+string attachmentId = "";
+
+@test:Config {
+    enable: true,
+    dependsOn: [tesCreateDraft]
+}
+function testListMessages() {
+    log:printInfo("oneDriveClient->listMessages()");
+    var output = oneDriveClient->listMessages(folderId = "drafts", optionalUriParameters = "?$select:\"sender,subject\"");
+    if (output is stream<Message, error?>) {
+        int index = 0;
+        error? e = output.forEach(function(Message queryResult) {
+            index = index + 1;
+            messageId = queryResult?.id.toString();
+        });
+        log:printInfo("Total count of records : " + index.toString());
+    } else {
+        test:assertFail(msg = output.toString());
+    }
+}
+
+@test:Config {
+    enable: true
+}
+function tesCreateDraft() {
+    log:printInfo("oneDriveClient->tesCreateDraft()");
+    DraftMessage draft = {
+        subject: "Did you see last night's game?",
+        importance: "Low",
+        body: {
+            "contentType": "HTML",
+            "content": "They were <b>awesome</b>!"
+        },
+        toRecipients: [
+            {
+            emailAddress: {
+                address: "AdeleV@contoso.onmicrosoft.com",
+                name: "Dhanushka"
+            }
+        }
+        ]
+    };
+    var output = oneDriveClient->createMessage(draft);
```

Comment:
Shall we use exact type here? Please check other places.

Reply:
This is only for test purposes. However, I'll improve test cases as you have suggested in a separate PR